corehq/apps/app_manager/management/commands/helpers.py

```diff
@@ -54,18 +56,51 @@ def add_arguments(self, parser):
             default=False,
             help="Perform the migration but don't save any changes",
         )
+        parser.add_argument(
+            '--use-chunks',
+            action='store_true',
+            default=False,
+            help='''To break up this management command across multiple chunks of domains to make it executable
+            over a large number of domains, set this to true and use the 'chunk-number' and 'total-chunks'
+            variables. Assumes that the result of get_domains() is ordered and constant between command
+            executions.''',
```

Comment:
I think Simon means write the list of domains to a text file.  That might look like:
* On first run, write the list of domains to a text file
* Iterate through that file, keeping track of the current position somewhere it can be referenced in the event of failure, say by printing to the console or updating another file with each new domain
* After a failure or ctrl+c, run the command again with an arg like `--skip-domains=379`, or just have the command look for a "progress" file and start there
* Iterate through the same domains file, skipping the first N domains

You could also probably update the file as you iterate through it, though I'm not certain how straightforward it is to modify a file while iterating over it.